model.py

```diff
@@ -3250,12 +3228,9 @@ def all_editions(self, recursion_level=5):
 
     def all_identifier_ids(self, recursion_level=5):
         _db = Session.object_session(self)
-        primary_identifier_ids = [
-            x.primary_identifier.id for x in self.editions
-            if x.primary_identifier
-        ]
+        primary_identifier_id = self.presentation_edition.primary_identifier.id
```

Comment:
The old code has to go but I don't think this is right. I think you now want to iterate over all the LicensePools associated with the Work and use their identifiers as your initial set.

If a Work has multiple LicensePools, we've made a determination (based on permanent work ID) that the Identifiers of those LicensePools identify "the same book". This is the sort of thing we usually track in `equivalencies`, and that's what `Identifier.recursively_equivalent_identifier_ids_flat` handles. But when we decide that two LicensePools identify the same Work, we don't actually create an equivalency between their Identifiers. So picking one of those Identifiers and using it as the initial set will tend to leave Identifiers out.

Maybe we should create those equivalencies! If we did that, the presentation edition's identifier ID would be equivalent to the identifier IDs of all the other LicensePools associated with that work. In that case, the only change you'd have to make here is to change `recursion_level` from 5 to 6 (because there's now an extra step between the identifier of the presentation edition and the identifier associated with one of the other LicensePools).

But I think it would be simpler to not create that equivalency, and generate our initial set of `primary_identifier_ids` from the identifiers of all the LicensePools grouped together in this Work.